### PR TITLE
Improve our flake8 linting tests

### PR DESCRIPTION
**Pull request by cclauss**
##### Checklist

- [ ] `npm install && npm test` passes
- [ ] tests are included 
- [ ] documentation is changed or added
- [ ] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

##### Description of change

This adds many more flake8 linting tests to align with `gyp-next`.  This PR ran ` black ./test/fixtures/test-charmap.py ./update-gyp.py` to format those files to align with https://github.com/psf/black formatting.

### CODE REVIEW

**Review by imatlopez — APPROVED** [review]

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -36,16 +36,10 @@ jobs:
           echo 'GYP_MSVS_OVERRIDE_PATH=C:\\Dummy' >> $Env:GITHUB_ENV
       - name: Lint Python
         if: matrix.os == 'ubuntu-latest'
-        run: |
-          # stop the build if there are Python syntax errors or undefined names
-          flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
-          # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
-          flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
+        run: flake8 . --ignore=E203,W503  --max-complexity=101 --max-line-length=88 --show-source --statistics
```
Comment (gengjiawen) [inline .github/workflows/tests.yml:39]: Should we keep max-line-length to 127 ?

Comment (cclauss) [reply .github/workflows/tests.yml:39]: 88 is the way to go... https://black.readthedocs.io/en/stable/the_black_code_style.html#line-length

**Review by gengjiawen — APPROVED** [review]

**Comment by gengjiawen** [comment]
@cclauss Can you do a force push in master with following commit message:

```
gyp: Improve our flake8 linting tests 

PR-URL: https://github.com/nodejs/node-gyp/pull/2356
Reviewed-By: Jiawen Geng <technicalcute@gmail.com>
```

We rely on structted commit msg to generate changelog before we got https://github.com/nodejs/node-gyp/pull/2330 merged.

**Comment by cclauss** [comment]
Done.

**Comment by gengjiawen** [comment]
> Done.

We need this on this repo :). Looks like you force push on your fork.

**Comment by cclauss** [comment]
git push --force origin
git push --force upstream

File: gyp/pylib/gyp/generator/msvs.py [hunk 2]
```diff
@@ -3276,9 +3276,9 @@ def GetEdges(node):
             # append to the default value. I.e. PATH=$(PATH);other_path
             edges.update(
                 {
-                        v
-                        for v in MSVS_VARIABLE_REFERENCE.findall(value)
-                        if v in properties and v != node
+                    v
+                    for v in MSVS_VARIABLE_REFERENCE.findall(value)
+                    if v in properties and v != node
```
[the remark this answers is not in the document]
Comment (cclauss) [reply gyp/pylib/gyp/generator/msvs.py:3281]: nodejs/gyp-next#102

**Pull request merged**